android/titanium/src/java/org/appcelerator/titanium/util/TiFileHelper.java

```diff
@@ -535,91 +528,79 @@ public void deployFromZip(File fname, File dest) throws IOException
 		}
 	}
 
-	public void wipeDirectoryTree(File path)
+	/**
+	 * Recursively deletes the given directory tree.
+	 * Will never throw an exception and will return the result as a boolean instead.
+	 * @param file Reference to a file or directory. Can be null.
+	 * @return
+	 * Returns true if successfully deleted all files and folders under given directory tree.
+	 * Returns false if at least 1 deletion failed or if given a null argument.
+	 */
+	public boolean tryDeleteTree(File file)
 	{
-		TreeSet<String> dirs = new TreeSet<String>(new Comparator<String>() {
-			public int compare(String o1, String o2)
-			{
-				return o1.compareTo(o2) * -1;
-			}
-		});
-
-		wipeDirectoryTree(path, dirs);
-
-		Iterator<String> d = dirs.iterator();
-		while (d.hasNext()) {
-			String fn = d.next();
-			File f = new File(fn);
-			Log.d(TAG, "Deleting Dir: " + f.getAbsolutePath(), Log.DEBUG_MODE);
-			f.delete();
+		boolean wasSuccessful = false;
```

Comment:
This is entirely a style thing, but personally I'd just do:
``` java
try {
  return deleteTree(file);
} catch (Throwable ex) {
  Log.e(TAG, "Failed to delete directory tree: " + file, ex);
}
return false;
```
I am not a fan of having temp result variables when you can directly return a result or literal.